sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/KinesisReader.java

```diff
@@ -142,4 +147,26 @@ public Instant getWatermark() {
     return source;
   }
 
+  /**
+   * Returns total size of all records that remain in Kinesis stream after current watermark.
+   * When currently processed record is not further behind than {@link #allowedRecordLateness}
+   * then this method returns 0.
+   */
+  @Override
+  public long getTotalBacklogBytes() {
+    Instant watermark = getWatermark();
+    if (watermark.plus(allowedRecordLateness).isAfterNow()) {
+      return 0L;
+    }
+    long backlogBytes = lastBacklogBytes;
+    try {
+      backlogBytes = kinesis.getBacklogBytes(source.getStreamName(), watermark);
```

Comment:
This method might be caused quite often. Since the stats are updated only once very minute, you could cache the values for a few seconds. Since getWatermark() returns `now`, does getBacklogBytes ever return non-zero?